specification/managementgroups/resource-manager/Microsoft.Management/stable/2020-02-01/management.json

```diff
@@ -0,0 +1,1997 @@
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "GetDescendants": {
+            "$ref": "./examples/GetDescendants.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions/{subscriptionId}": {
+      "put": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "ManagementGroupSubscriptions_Create",
+        "description": "Associates existing subscription with the management group.\n",
+        "parameters": [
+          {
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CacheControlHeader"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+		  "204": {
+            "description": "NoContent"
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AddSubscriptionToManagementGroup": {
+            "$ref": "./examples/AddManagementGroupSubscription.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "ManagementGroupSubscriptions_Delete",
+        "description": "De-associates subscription from the management group.\n",
+        "parameters": [
+          {
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CacheControlHeader"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+		  "204": {
+            "description": "NoContent"
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DeleteSubscriptionFromManagementGroup": {
+            "$ref": "./examples/RemoveManagementGroupSubscription.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{groupId}/settings": {
+      "get": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "HierarchySettings_List",
```

Comment:
>HierarchySettings_List [](start = 24, length = 22)

it is just /Settings.. Do you want to rename resource type to hierarchySettings ?

or rename the operation to match the type

Reply:
What do you mean rename the operation to match the type? We want to use /settings in our URI, but the resource name to be hierarchy settings. Is there something wrong with having different names or is this more of a NIT?

Reply:
'resource name' ?

Operation id format is  'ResourceType_operation'
It is not entirely nit.. But if that is how you want SDK to be generated, it is fine. 

---
In reply to: [389166229](https://github.com/Azure/azure-rest-api-specs/pull/8443#discussion_r389166229) [](ancestors = 389166229)

Reply:
Oh, now I understand your question. The resource type is "hierarchySettings" and the operation type is "List" (as it's a collection Get), thus the operation Id. Do you see anything wrong with it?